main.cpp

```diff
@@ -122,394 +172,476 @@ struct CarWash
     
-    //attenuate signal
     void attenuate(int coefficient = -50);
-    //consume power
     void inputPower(int amountOfPower = 20);
-    //recieve control signal
-    int readKnob (double knobAngle = 10.123);
+    int readKnob (int knobAngle = 10);
 };
-/*
-Thing 8) headphone amplifier
-5 properties:
-    1) SNR
-    2) output power
-    3) output impendance
-    4) frequency range
-    5) input voltage 
-3 things it can do:
-    1) input signal
-    2) amplify signal
-    3) produce noise
- */
+
 struct HeadphoneAmp
 {
-    //SNR
     int SNR = 112;
-    //output power
     float outPower = 600.0f;
-    //output impendance
     int outImpendance = 32;
-    //frequency range
     int maxFreq = 20000;
-    //input voltage
     double inVolt = 12;
     
-    //input signal
     void getInput(int channelNum = 0);
-    //amplify signal
     void doAmp(int channelNum = 0);
-    //produce noise
     bool noiseReduction(int filterNum = 3);
 };
-/*
-Thing 9) body
-5 properties:
-    1) height
-    2) width
-    3) depth
-    4) colour
-    5) material
-3 things it can do:
-    1) hold internal parts
-    2) protect circuit
-    3) collect dust
- */
+
 struct Body
 {
-    //height
     float height = 6.5f;
-    //width
     float width = 15.0f;
-    //depth
     float depth = 10.5f;
-    //colour
     char colour = 'B';
-    //material
     char material = 'W';
     
-     //hold internal parts
-     bool checkTheBolt(int circuitId = 1);
-     //protect circuit
-     bool checkShortCircuit(int circuitId = 1);
-     //collect dust
-     void alarmOverDust(float time = 10);
+    bool checkTheBolt(int circuitId = 1);
+    bool checkShortCircuit(int circuitId = 1);
+    void alarmOverDust(float time = 10);
 };
-/*
-Thing 10) Audio monitor controller
-5 properties:
-    1) DAC
-    2) power unit
-    3) VCA
-    4) headphone amplifier
-    5) body
-3 things it can do:
-    1) control volume
-    2) set input sources
-    3) toggle crosfeed effect
- */
 
 struct MonitorController
 {
-    //left channel DAC
     DAC DACLeft;
-    //right channel DAC
     DAC DACRight;
-    //power supply unit
     PowerUnit PSU0;
-    //volt controlled attenuator
     VCA VCA0;
-    //left channle amplifier
     HeadphoneAmp AmpLeft;
-    //right channle amplifier
     HeadphoneAmp AmpRight;
-    //body of controller
     Body body0;
 
-    //control volume
     void setVol(int amount = 0);
-    //set input sources
     bool selectSource(int sourceNum, bool status=0);
-    //toggle crosfeed effect
     bool toggleCrossfeed(bool status = 0);
 };
-   
+
+void CanPlant::OfficeWorker::makeCall(int customerId, float time)
+{
+    SipProvider sipnet;
+    sipnet.makeCall(intNumber, customerId);
+    time++;
+}
+
+void CanPlant::OfficeWorker::pickCall(int number, float time)
+{
+    //int recordedNumber = number;
+    number = 0;
+    time++;
+}
+
+int CanPlant::OfficeWorker::reserveGood(int customerId, int amount)
+{
+    customerId++;
+    return amount++;
+}
```

Comment:
You should always prefer to use pre-increment/decrement instead of post-increment/decrement
also this line:
```
return amount++;
```
this will basically do this behind the scenes:
```
auto result = amount;
amount += 1;
return result;
```
it's visually ambiguous what value will be returned when you use post-increment in the return statement.
- [ ] revise.